### PR TITLE
Issue Template: @grafana/ui component request

### PR DESCRIPTION
**Pull request by tskarhed**
**What this PR does / why we need it**:
Adds a new issue template with a component request for @grafana/ui. The checklist will be used to make sure that the component is at 100%

All feedback would be greatly appreciated!

**Special notes for your reviewer**:
Not sure if I did the frontmatter correctly.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/4-grafana_ui_component.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+name: '@grafana/ui component request'
+about: Suggest a component for the @grafana/ui package
+labels: 'area/grafana/ui'
+---
+
+<!-- 
+By using this template you will make it easeier for us to make sure that documentation and implementation stays up to date for every component in @grafana/ui
```
Comment (oddlittlebird) [inline .github/ISSUE_TEMPLATE/4-grafana_ui_component.md:8]:
```suggestion
By using this template you will make it easier for us to make sure that documentation and implementation stays up to date for every component in @grafana/ui.
```

File: .github/ISSUE_TEMPLATE/4-grafana_ui_component.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+Thank you!
+-->
+
+**Why this component is needed**:
+<!-- Explain your use case -->
+___
+ - [ ] Is/could it be used enough to become a pattern?
```
Comment (oddlittlebird) [inline .github/ISSUE_TEMPLATE/4-grafana_ui_component.md:16]:
```suggestion
 - [ ] Is/could it be used in more than one place in Grafana?
```

File: .github/ISSUE_TEMPLATE/4-grafana_ui_component.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+ **Where is/could it be used?**:
+ Post screenshots if you have any.
```
Comment (oddlittlebird) [inline .github/ISSUE_TEMPLATE/4-grafana_ui_component.md:19]: I suggest moving line 19 under the checklist.

File: .github/ISSUE_TEMPLATE/4-grafana_ui_component.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+___
+- [ ] It has a single use case
+- [ ] It is/could be used in multiple places
+
+**Implementation** (Checklist meant for developer)
```
Comment (oddlittlebird) [inline .github/ISSUE_TEMPLATE/4-grafana_ui_component.md:25]: Is the "developer" here the contributing developer or the maintainer developer? I suggest making this extremely clear.

File: .github/ISSUE_TEMPLATE/4-grafana_ui_component.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+- [ ] Component has a story in Storybook
+- [ ] Props and naming follows [our style guide]()
```
Comment (oddlittlebird) [inline .github/ISSUE_TEMPLATE/4-grafana_ui_component.md:28]: We have several style guides that could apply to UI. Please use the style guide name as the link text.

File: .github/ISSUE_TEMPLATE/4-grafana_ui_component.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+- [ ] It is extendable (eg. rest props are spreaded, styles with className works)
+- [ ] Works with both light and dark theme
+- [ ] Uses theme for spacing, colors etc.
```
Comment (oddlittlebird) [inline .github/ISSUE_TEMPLATE/4-grafana_ui_component.md:31]:
```suggestion
- [ ] Uses theme for spacing, colors, and so on
```

File: .github/ISSUE_TEMPLATE/4-grafana_ui_component.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+___
+- [ ] It has a single use case
+- [ ] It is/could be used in multiple places
+
+**Implementation** (Checklist meant for developer)
+
+- [ ] Component has a story in Storybook
+- [ ] Props and naming follows [our style guide]()
+- [ ] It is extendable (eg. rest props are spreaded, styles with className works)
```
Comment (oddlittlebird) [inline .github/ISSUE_TEMPLATE/4-grafana_ui_component.md:29]:
```suggestion
- [ ] It is extendable (rest props are spreaded, styles with className works, and so on)
```

File: .github/ISSUE_TEMPLATE/4-grafana_ui_component.md (hunk 6 above)
Comment (oddlittlebird) [reply .github/ISSUE_TEMPLATE/4-grafana_ui_component.md:31]: A link to the Theming style guide might be a good idea here.

File: .github/ISSUE_TEMPLATE/4-grafana_ui_component.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+- [ ] Works with both light and dark theme
+- [ ] Uses theme for spacing, colors etc.
+
+**Documentation**
+
+- [ ] Properties are documented
+- [ ] Use cases are described
+- [ ] Code exmaples for the different use cases
```
Comment (oddlittlebird) [inline .github/ISSUE_TEMPLATE/4-grafana_ui_component.md:37]:
```suggestion
- [ ] Code examples for the different use cases
```

**Review by oddlittlebird — CHANGES_REQUESTED** [review]
Good stuff

File: .github/ISSUE_TEMPLATE/4-grafana_ui_component.md (hunk 6 above)
Comment (dprokop) [inline .github/ISSUE_TEMPLATE/4-grafana_ui_component.md:31]: Link to theming would be awesome here :)

**Review by dprokop — APPROVED** [review]
Think this is a great start @tskarhed!

**Review by jessover9000 — COMMENTED** [review]
some great feedback from @oddlittlebird , other than that I think it's great :) @peterholmberg or @Clarity-89 could you take care of getting this changed up and merged?

File: .github/ISSUE_TEMPLATE/4-grafana_ui_component.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+**Why is this component needed**:
+<!-- Explain your use case -->
+___
+ - [ ] Is/could it be used in more than one place in Grafana?
+
+**Where is/could it be used?**:
+
+___
+- [ ] Post screenshots possible.
+- [ ] It has a single use case
```
Comment (oddlittlebird) [inline .github/ISSUE_TEMPLATE/4-grafana_ui_component.md:22]: Put periods at the end of all sentences. In this case, all options.

File: .github/ISSUE_TEMPLATE/4-grafana_ui_component.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+- [ ] It is/could be used in multiple places
+
+**Implementation** (Checklist meant for the person implementing the component)
+
+- [ ] Component has a story in Storybook
+- [ ] Props and naming follows [our style guide](https://github.com/grafana/grafana/blob/master/contribute/style-guides/frontend.md)
+- [ ] It is extendable (rest rest props are spreaded, styles with className works, and so on)
```
Comment (oddlittlebird) [inline .github/ISSUE_TEMPLATE/4-grafana_ui_component.md:29]:
```suggestion
- [ ] It is extendable (rest props are spread, styles with className works, and so on).
```

**Review by oddlittlebird — CHANGES_REQUESTED** [review]

**Review by oddlittlebird — APPROVED** [review]
Looks good!

**Pull request merged**